### PR TITLE
feat(forms): support hidden widget for object type

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Form does not support `"ui:widget" : "hidden"` for field type `object`

**What is the new behavior?**
Form supports this feature

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

File: packages/forms/src/fields/ObjectField.js [hunk 1]
```diff
@@ -103,6 +103,9 @@ class ObjectField extends Component {
 		const { widget, ...options } = getUiOptions(uiSchema);
 		// widget
 		if (typeof widget === 'string') {
+			if (widget === 'hidden') {
+				return (<input type='hidden'/>);
```
Comment (talend-glorieux) [inline packages/forms/src/fields/ObjectField.js]: I'm not sure that implementation will actually work.
The main issue is that an object on the JsonSchema doesn't represent an actual input. I'm not sure if it's a good idea to replace a collection of inputs by by single hidden one.

Comment (talend-glorieux) [inline packages/forms/src/fields/ObjectField.js]: There is also the fact the you are not using the library's HIddenWidget.

Comment (frassinier) [inline packages/forms/src/fields/ObjectField.js]: My proposal was if `ui:widget` set to hidden for Object type, then it should render nothing, wha do you think?

File: output/forms.eslint.txt [hunk 2]
```diff
@@ -103,6 +103,10 @@ The react/require-extension rule is deprecated. Please use the import/extensions
 /home/travis/build/Talend/ui/packages/forms/src/fields/BooleanField.js
   25:3  error  'name' is missing in props validation  react/prop-types
 
+/home/travis/build/Talend/ui/packages/forms/src/fields/ObjectField.js
```
Comment (frassinier) [inline output/forms.eslint.txt]: Please, fix these lint errors

File: packages/forms/stories/json/hidden.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        }
+      }
+    }
+  },
+  "uiSchema": {
+    "talendTab": {
+      "ui:widget": "hidden",
+      "main": {
+          "password": {
+            "ui:widget": "password"
```
Comment (frassinier) [inline packages/forms/stories/json/hidden.json]: Useless since the main object is hidden

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

File: packages/forms/src/fields/ObjectField.js [hunk 4]
```diff
@@ -103,6 +105,9 @@ class ObjectField extends Component {
 		const { widget, ...options } = getUiOptions(uiSchema);
 		// widget
 		if (typeof widget === 'string') {
+			if (widget === 'hidden') {
+				return (<HiddenWidget />);
```
Comment (frassinier) [inline packages/forms/src/fields/ObjectField.js]: Why do you return input here? Please, try to render nothing!
The goal is to hide this object :)

Comment (charlesnguyen) [inline packages/forms/src/fields/ObjectField.js]: It is to be consistent with other widget types.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

**Comment by sgandon** [comment]
Hello guys, what is the status of this PR.
I can see 2 benefit for this enhancement, a faster serialization because all the hidden sub-elements are not generated and also a smaller payload (because of the previous reason) to be transiting over the wire.

**Comment by frassinier** [comment]
@glorieux wants to test it before approving

> all the hidden sub-elements are not generated

I do not get it, what did you mean?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

**Comment by sgandon** [comment]
for the sake of history 
this is more compact
to have this
```
"uiSchema": {
  "talendTab": {
    "ui:widget": "hidden"
  }
}
```
than this
```
"uiSchema": {
  "talendTab": {
    "main": {
       "login": {
          "ui:widget": "hidden"
       }
       "firstName": {
          "ui:widget": "hidden"
       }
       "lastName": {
          "ui:widget": "hidden"
       }
       "password": {
          "ui:widget": "hidden"
       }
       "password2": {
          "ui:widget": "hidden"
       }
    }
  }
}
```

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://250.talend.surge.sh)

**Pull request merged**